src/shogun/regression/LeastAngleRegression.cpp

```diff
@@ -122,7 +124,7 @@ bool LeastAngleRegression::train_machine_templated(const std::shared_ptr<DenseFe
 	m_is_active.resize(n_fea);
 	fill(m_is_active.begin(), m_is_active.end(), false);
 
-	SGVector<ST> y = regression_labels(m_labels)->template get_labels_t<ST>();
+	SGVector<ST> y = labs->as<RegressionLabels>()->template get_labels_t<ST>();
```

Comment:
shoudn't this be `regression_labels(labs)`? @karlnapf this does the implicit conversion from one label type to another if possible right?

Reply:
it should be exactly this ... but the label encoder needs to do its work for this to work...I think you can just leave it like this if nothing breaks

Reply:
but the label encoder does not handle `RegressionLabels`. Should it?

Reply:
probably not. If a user passed discrete labels to a regression algorithm, she will just see this error here (not liky)

Reply:
I guess we could imagine the case where we have two different kind of regression labels ... then we wouldnt want to assert the right subclass here if they are convertable. So actually I would pass *all* labels through the label encoder in `train` ... and the label encoder checks the a class is already of the right type and satisfies all the constraints (none here in this case) and if so, it just forwards.
So I think YES, it should be dealt with by the label encoder